public/content/en/vibed/deploy-on-heroku.md

```diff
@@ -0,0 +1,188 @@
+
+```d
+shared static this() {
+  // ...
+  auto settings = new HTTPServerSettings;
+  // Provide a default port in case of the $PORT variable isn't set.  
+  settings.port = environment.get("PORT", "8080").to!ushort;
+  listenHTTP(settings, router);
+}
+```
+
+Also create a `Procfile`, which is a text file in the root directory of the application, which explicitly declare what command 
+should be executed to start the app.
+
+The Procfile in the example app looks like this:
+
+```
+web: ./hello-world
+```
+
+### 2 : Prepare the app 
+
+Before going further login to the heroku cli.
+
+To login use the [heroku toolbelt](https://toolbelt.heroku.com/standalone).
+
+This provides access to the Heroku Command Line Interface (CLI), which can be used for managing and scaling your applications and add-ons.
+
+After installing the toolbet just run the following
+
+```
+$ heroku login
+```
+
+### 3 : Create the app 
+
+To do so, go on the [heroku dashboard](https://dashboard.heroku.com) and create a new app. 
+After doing this memorize the name of the created app, it will be useful later. 
+
+or use the command line like this 
+
+```
+$ heroku create
+Creating app... done, ⬢ rocky-hamlet-67506
+https://rocky-hamlet-67506.herokuapp.com/ | https://git.heroku.com/rocky-hamlet-67506.git
+```
+
+The app's name here is *rocky-hamlet-67506*. 
+
+### Deploy using git 
+
+To deploy the app directly use `git` commands. A separate git remote endpoint should be added to which new releases can be pushed.
+
+As shown in the previous section, our name's app is rocky-hamlet-67506. But change it to yours. 
+
+```
+$ heroku git:remote -a rocky-hamlet-67506
+```
+
+Notice the remote endpoint is added to the git config
+
+```
+$ git remote -v
+heroku	https://git.heroku.com/rocky-hamlet-67506.git (fetch)
+heroku	https://git.heroku.com/rocky-hamlet-67506.git (push)
+```
+
+### Adding the buildpack
+
+Buildpacks are responsible for transforming deployed code into a slug, 
+which can then be executed on a dyno. Buildpacks are composed of a set of scripts, 
+and depending on the programming language, the scripts will retrieve dependencies, 
+output generated assets or compiled code, and more.
+
+For more information browse the [Heroku documentation](https://devcenter.heroku.com/articles/buildpacks)
+
+To deploy we are using this [webpack](https://github.com/MartinNowak/heroku-buildpack-d), the heroku-buildpack-vibe.d. 
+
+```
+$ heroku buildpacks:set https://github.com/MartinNowak/heroku-buildpack-d
+```
+By default the buildpack uses the latest dmd compiler. 
+It is possible to use gdc or ldc and to choose a specific compiler versions by adding a .d-compiler file to your project. 
+Use dmd, ldc, or gdc to select the latest or dmd-2.068.2, ldc-0.16.0, or gdc-4.9.2 to 
+select a specific version of a compiler.
+
+### Deploy the code 
+
+Proceed in your usual git habit and write awesome code. 
+
+To release a new version, just push the newest version to the heroku endpoint.
```

Comment:
A new version can easily be released by solely pushing the version to the __Heroku__ endpoint: